lib/internal/http.js

```diff
@@ -31,9 +32,30 @@ function ondrain() {
   if (this._httpMessage) this._httpMessage.emit('drain');
 }
 
+function now() {
+  const hr = process.hrtime();
+  return hr[0] * 1000 + hr[1] / 1e6;
+}
+
+class HttpRequestTiming extends PerformanceEntry {
+  constructor(statistics) {
+    super();
+    this.name = 'HttpRequest';
+    this.entryType = 'http';
+    this.duration = statistics.endTime - statistics.startTime;
+    this.startTime = statistics.startTime;
+  }
+}
+
+function emitStatistics(statistics) {
+  notify('http', new HttpRequestTiming(statistics));
+}
+
 module.exports = {
   outHeadersKey: Symbol('outHeadersKey'),
   ondrain,
   nowDate,
-  utcDate
+  utcDate,
+  now,
```

Comment:
I think this would be better exported by some other internal module as it's not http-specific.

Reply:
I didn't find any internal module that could contain this, do you think it's safe to add a new one ?

Reply:
The existing lib/internal/util.js is probably the best fit.

Reply:
Actually, since only one file (lib/_http_server.js) is using the function, maybe it would be better to just move the definition to that file.